### PR TITLE
[codex] Fix marker resolution and duplicate mapping validation

### PR DESCRIPTION
**Pull request by satvik007**
Resolves: https://github.com/Hypersequent/tms-issues/issues/2442

## Summary

This branch fixes the remaining result-mapping issues in `qas-cli` by making parsed marker metadata the source of truth for upload mapping and by validating duplicate target mappings earlier in the flow.

The branch also bumps the package version to `0.6.1`.

## Root Cause

The previous uploader logic inferred identity from `result.name`, which is a display field and may include Playwright `describe(...)` titles. When those describe titles contained multiple QA Sphere markers, the uploader could map a result to the wrong run test case simply because that marker appeared earlier in the string.

There was also a second safety issue after duplicate-target detection was introduced: duplicate validation could still happen after side effects had already occurred. The initial fix moved that check ahead of run creation, but `--create-tcases` could still create new test cases before the duplicate rejection fired.

## What Changed

- Added explicit marker metadata and resolution state to `TestCaseResult`
- Added shared mapping helpers so handler and uploader use the same marker-based mapping rules
- Switched uploader mapping from name scanning to explicit marker resolution
- Updated Playwright parsing to:
  - fan out one result per annotation marker
  - extract fallback markers from `spec.title` only, not `describe` prefixes
  - preserve `resolved-none` when only the `describe` title contains markers
- Updated JUnit and Allure parsing to populate marker metadata directly where possible
- Updated handler resolution logic so parser decisions are authoritative:
  - `resolved` stays resolved
  - `resolved-none` is not re-parsed from display names
  - `needs-project-resolution` still allows the project-aware JUnit fallback path
- Added duplicate-target detection based on resolved target test cases instead of string matching
- Kept `--force` as the override for duplicate-target failures
- Moved duplicate validation ahead of side effects for implicit uploads:
  - before run creation
  - before `--create-tcases` creation calls
- Split `--create-tcases` handling into a read-only planning step and a post-validation creation/finalization step
- Updated CLI/README text so `--force` and mapping behavior describe duplicate-target handling explicitly
- Bumped `package.json` and root `package-lock.json` metadata to `0.6.1`

## Why Reviewers Should Care

This is primarily a correctness fix.

Without these changes, a Playwright suite could upload results to the wrong QA Sphere test cases even when the result count looked plausible. In failure cases, the CLI could also leave behind unwanted artifacts:

- orphaned runs from post-create duplicate rejection
- orphaned test cases from `--create-tcases` when duplicate validation failed too late

The branch makes parser intent explicit and ensures duplicate rejection happens before either of those side effects.

## Suggested Review Order

1. `src/utils/result-upload/types.ts`
2. `src/utils/result-upload/mapping.ts`
3. `src/utils/result-upload/parsers/playwrightJsonParser.ts`
4. `src/utils/result-upload/ResultUploadCommandHandler.ts`
5. `src/utils/result-upload/ResultUploader.ts`
6. `src/tests/playwright-json-parsing.spec.ts`
7. `src/tests/result-upload.spec.ts`

## Tests Added

- Playwright regression where `describe` titles contain unrelated markers but `spec.title` contains the real one
- Playwright regression where only the `describe` title contains markers and the result must remain unmatched
- Upload regression proving duplicate-target mappings fail before creating a new run
- Upload regression proving duplicate-target mappings fail before creating new test cases under `--create-tcases`
- Existing duplicate-target and force-path coverage retained

## Validation

- [x] `npm run check`
- [x] `npm test`

## Notes

- Follow-up package version in this branch: `0.6.1`

### CODE REVIEW

File: src/tests/playwright-json-parsing.spec.ts [hunk 1]
```diff
@@ -509,6 +517,118 @@ describe('Playwright JSON parsing', () => {
...
 		expect(testcases[0].name).toBe('PRJ-100: PRJ-999: Test with marker in name')
 		expect(testcases[1].name).toBe('PRJ-200: PRJ-999: Test with marker in name')
+		expect(testcases[0].marker).toEqual({ projectCode: 'PRJ', seq: 100 })
+		expect(testcases[1].marker).toEqual({ projectCode: 'PRJ', seq: 200 })
+		expect(testcases[0].markerResolution).toBe('resolved')
+		expect(testcases[1].markerResolution).toBe('resolved')
+	})
+
+	test('Should extract fallback marker from spec title, not describe titles', async () => {
+		const jsonContent = JSON.stringify({
```
Comment (ramilamparo) [inline src/tests/playwright-json-parsing.spec.ts:527]: The newly created JSON files is the same as this. Import it instead.

File: src/tests/result-upload.spec.ts [hunk 2]
```diff
@@ -674,6 +688,112 @@ describe('Multi-annotation Playwright upload', () => {
...
+	})
+
+	test('Should leave Playwright results unmatched when only describe titles contain markers', async () => {
+		const numResultUploadCalls = countResultUploadApiCalls()
+
+		await expect(
+			run(
+				`playwright-json-upload -r ${runURL} ./src/tests/fixtures/playwright-json/describe-marker-unmatched.json`
+			)
+		).rejects.toThrowError()
```
Comment (ramilamparo) [inline src/tests/result-upload.spec.ts:708]: I'm getting deprecation notices, use `toThrow` instead.

**Review by ramilamparo — APPROVED** [review]
Looks good. This edge case makes sense to me.

File: src/utils/result-upload/ResultUploadCommandHandler.ts [hunk 3]
```diff
@@ -446,7 +557,83 @@ export class ResultUploadCommandHandler {
...
+			return null
+		}
+
+		result.marker = {
+			projectCode: this.markerParser.detectProjectCode(result.name) || projectCode,
+			seq,
+		}
+		result.markerResolution = 'resolved'
+		return result.marker
+	}
```
Comment (ramilamparo) [inline src/utils/result-upload/ResultUploadCommandHandler.ts:625]: I think this is still the job of parsers. If we move it to the parsers, we can simplify the code a bit. We can drop `markerResolution` field and derive this information whether `marker` field is null or not.

But I understand that the parsers (specifically JUnit) needs context about the project code to correctly derive marker from the test name. We can skip it for now.
